integrations/operator/controllers/resources/teleport_reconciler.go

```diff
@@ -135,19 +140,20 @@ func (r TeleportResourceReconciler[T, K]) Upsert(ctx context.Context, obj kclien
 
 	teleportResource.SetOrigin(types.OriginKubernetes)
 
-	// Propagate revision as required by opportunistic locking
-	if exists {
-		teleportResource.SetRevision(existingResource.GetRevision())
-	}
-
-	// We apply resource-specific mutations.
-	if mutator, ok := r.resourceClient.(TeleportResourceMutator[T]); ok {
-		mutator.Mutate(teleportResource, existingResource)
-	}
-
 	if !exists {
+		// This is a new resource
+		if mutator, ok := r.resourceClient.(TeleportResourceMutator[T]); ok {
+			mutator.Mutate(teleportResource)
+		}
```

Comment:
Should we handle a `!ok` branch and return a proper error?

Reply:
Not all resources need mutations. I think it's easier to make mutation optional and continue, the alternative is to make mutation mandatory and add a bunch of empty functions to all resource clients.